commcare_connect/program/tests/test_views.py

```diff
@@ -0,0 +1,295 @@
+from http import HTTPStatus
+
+import pytest
+from django.contrib import messages
+from django.http import HttpResponseRedirect
+from django.test import Client
+from django.urls import reverse
+
+from commcare_connect.opportunity.tests.factories import DeliveryTypeFactory
+from commcare_connect.organization.models import Organization
+from commcare_connect.program.models import ManagedOpportunityApplication, ManagedOpportunityApplicationStatus, Program
+from commcare_connect.program.tests.factories import (
+    ManagedOpportunityApplicationFactory,
+    ManagedOpportunityFactory,
+    ProgramFactory,
+)
+from commcare_connect.users.models import User
+
+
+class BaseProgramTest:
```

Comment:
It would be good to add tests for the forms, I know the managed opportunity one handles a lot, and especially any validation conditions so both good and bad data. It is great that this includes a lot of happy path tests, but tests that confirm expected failures can also be very useful.

Reply:
Added